src/telemetry/Manager.cc

```diff
@@ -38,6 +69,415 @@ void Manager::InitPostBrokerSetup(broker::endpoint& ep)
 	pimpl.swap(ptr);
 	}
 
+// -- collect metric stuff -----------------------------------------------------
+
+template <typename T>
+zeek::RecordValPtr Manager::GetMetricOptsRecord(Manager::MetricType metric_type,
+                                                const broker::telemetry::metric_family_hdl* family)
+	{
+	static auto string_vec_type = zeek::id::find_type<zeek::VectorType>("string_vec");
+	static auto metric_opts_type = zeek::id::find_type<zeek::RecordType>("Telemetry::MetricOpts");
+
+	static auto prefix_idx = metric_opts_type->FieldOffset("prefix");
+	static auto name_idx = metric_opts_type->FieldOffset("name");
+	static auto help_text_idx = metric_opts_type->FieldOffset("help_text");
+	static auto unit_idx = metric_opts_type->FieldOffset("unit");
+	static auto is_total_idx = metric_opts_type->FieldOffset("is_total");
+	static auto labels_idx = metric_opts_type->FieldOffset("labels");
+	static auto bounds_idx = metric_opts_type->FieldOffset("bounds");
+	static auto metric_type_idx = metric_opts_type->FieldOffset("metric_type");
+
+	if ( metric_opts_cache.count(family) != 0 )
+		return metric_opts_cache[family];
+
+	auto r = make_intrusive<zeek::RecordVal>(metric_opts_type);
+	r->Assign(prefix_idx, make_intrusive<zeek::StringVal>(broker::telemetry::prefix(family)));
+	r->Assign(name_idx, make_intrusive<zeek::StringVal>(broker::telemetry::name(family)));
+	r->Assign(help_text_idx, make_intrusive<zeek::StringVal>(broker::telemetry::helptext(family)));
+	r->Assign(unit_idx, make_intrusive<zeek::StringVal>(broker::telemetry::unit(family)));
+	r->Assign(is_total_idx, val_mgr->Bool(broker::telemetry::is_sum(family)));
+
+	auto label_names_vec = make_intrusive<zeek::VectorVal>(string_vec_type);
+	for ( const auto& l : broker::telemetry::label_names(family) )
+		label_names_vec->Append(make_intrusive<StringVal>(l));
+	r->Assign(labels_idx, label_names_vec);
```

Comment:
For readbility, add empty line after block

```suggestion

	r->Assign(labels_idx, label_names_vec);
```